myfitnesspal/client.py

```diff
@@ -939,42 +936,39 @@ def set_new_food(
             url,
             data,
         )
-        if result.status_code == 200:
-            document = lxml.html.document_fromstring(result.content.decode("utf-8"))
+        if not result.ok:
```

Comment:
I've here switched around the order in which we handle the response so we "fail fast" and thus can remove a layer of indentation in the successful case.  I _believe_ this is equivalent, but please double-check me.

Reply:
Looks good :)